srv/salt/_modules/cephprocesses.py

```diff
@@ -49,101 +48,326 @@
 absent_processes = {'igw': ['lrbd']}
 
 
-def check(results=False, quiet=False, **kwargs):
+# pylint: disable=too-few-public-methods
+class SystemdUnit(object):
+    """
+    Class representation of the processes' systemd state
+    """
 
+    def __init__(self, proc_name=None, osd_id=None):
+        self.proc_name = proc_name
+        self._is_disabled = False
+        self.osd_id = osd_id
+        self.service_names = self._service_names()
+
+    @property
+    def is_disabled(self):
+        """
+        Reach out to systemctl and call is-enabled.
+        Property for this state
+        """
+        if not self.service_names:
+            return False
+
+        for fsn in self.service_names:
```

Comment:
I'd be really happy about a comment that this loop is here for nfs-ganesha, since it depends(?) on rpcbind. would it maybe be better to check for rpcbind if service_name == nfs-ganesha?

Reply:
Well, true. This check loop-check is mainly there for nfs-ganesha, but maybe in the future we'll have services that are dependent on more than just one service aswell..

> would it maybe be better to check for rpcbind if service_name == nfs-ganesha?

nfs-ganesha _and_ rpcbind have a systemd unit respectively. Imo we need to check both.

Reply:
> Well, true. This check loop-check is mainly there for nfs-ganesha, but maybe in the future we'll have services that are dependent on more than just one service aswell..

Like what exactly?

> > would it maybe be better to check for rpcbind if service_name == nfs-ganesha?
> 
> nfs-ganesha _and_ rpcbind have a systemd unit respectively. Imo we need to check both.

Thats what I mean. Drop the loop and add the condition to additionally check rpcbind if service_name == nfs-ganesha

Reply:
> Like what exactly?

What will the monitoring stack look like once it's a separate role? Should we check grafana & prometheus ? That's probably the only example I can think of right now.





> Thats what I mean. Drop the loop and add the condition to additionally check rpcbind if service_name == nfs-ganesha

The only 'con' to that suggestion would just be style. With the current design it's easy to add an arbitrary number of dependencies to a service whilst we have to catch (maybe eventually occuring) exceptions to the is_disabled property.

If we can't find more examples than nfs-ganesha I'm happy to change the code towards your suggestion